Doc/whatsnew/3.12.rst

```diff
@@ -1090,16 +1162,24 @@ Deprecated
   Python 3.14, when ``'data'`` filter will become the default.
   See :ref:`tarfile-extraction-filter` for details.
 
-* :mod:`typing`: :class:`typing.Hashable` and :class:`typing.Sized` aliases for :class:`collections.abc.Hashable`
-  and :class:`collections.abc.Sized`. (:gh:`94309`.)
+* :mod:`typing`:
+
+  * :class:`typing.Hashable` and :class:`typing.Sized` aliases for :class:`collections.abc.Hashable`
+    and :class:`collections.abc.Sized`. (:gh:`94309`.)
+
+  * :class:`typing.ByteString`, deprecated since Python 3.9, now causes a
+    :exc:`DeprecationWarning` to be emitted when it is used.
+    (Contributed by Alex Waygood in :gh:`91896`.)
 
 * :mod:`xml.etree.ElementTree`: The module now emits :exc:`DeprecationWarning`
   when testing the truth value of an :class:`xml.etree.ElementTree.Element`.
   Before, the Python implementation emitted :exc:`FutureWarning`, and the C
   implementation emitted nothing.
+  (Contributed by Jacob Walls in :gh:`83122`.)
 
-* The 3-arg signatures (type, value, traceback) of :meth:`~coroutine.throw`,
-  :meth:`~generator.throw` and :meth:`~agen.athrow` are deprecated and
+* The 3-arg signatures (type, value, traceback) of :meth:`coroutine throw()
+  <coroutine.throw>`, :meth:`generator throw() <generator.throw>` and
+  :meth:`async generator throw() <agen.athrow>` are deprecated and
```

Comment:
Added explicit titles to avoid the text saying ``throw()``, ``throw()``, and ``athrow()``